src/js/components/nft/nft-collection-view.jsx

```diff
@@ -0,0 +1,258 @@
+import * as format from '../../helper/format';
+import * as svg from '../../helper/svg';
+import ModalView from '../utils/modal-view';
+import * as text from '../../helper/text';
+import {changeLoaderState} from '../loader';
+import Transaction from '../../common/transaction';
+import async from 'async';
+import utils from '../../helper/utils';
+
+
+class NftCollectionView extends Component {
+    constructor(props) {
+        super(props);
+        this.datatable_reload_interval = undefined;
+        this.state                     = {
+            nft_list                    : [],
+            nft_selected                : undefined,
+            datatable_reload_timestamp  : '',
+            datatable_loading           : false,
+            modal_show_burn_confirmation: false,
+            modal_show_burn_result      : false,
+            modal_burn_create_asset     : true,
+            modal_body_burn_result      : []
+        };
+    }
+
+    componentDidMount() {
+        this.reloadCollection();
+        this.datatable_reload_interval = setInterval(() => this.reloadCollection(), 60000);
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.datatable_reload_interval);
+        this.state.nft_list.forEach(nft => nft.src && URL.revokeObjectURL(nft.src));
+    }
+
+    reloadCollection() {
+        this.setState({
+            datatable_loading: true
+        });
+
+        return API.listTransactionWithDataReceived(this.props.wallet.address_key_identifier, 'tangled_nft').then(data => {
+            async.mapLimit(data, 6, (row, callback) => {
+                utils.getImageFromApi(row)
+                    .then(imageUrl => callback(null, {
+                        src   : imageUrl,
+                        width : 4,
+                        height: 3,
+                        hash  : row.transaction_output_attribute[0].value.file_list[0].hash,
+                        amount: row.amount,
+                        txid  : row.transaction_id
+                    }));
+            }, (err, nftList) => {
+                this.setState({
+                    nft_list                  : nftList,
+                    datatable_reload_timestamp: new Date(),
+                    datatable_loading         : false
+                });
+            });
+        });
+    }
+
+    cancelNftBurn() {
+        this.setState({
+            nft_selected                : undefined,
+            modal_show_burn_confirmation: false,
+            modal_burn_create_asset     : true
+        });
+    }
+
+
+    prepareTransactionOutputToBurnNft(nft, keepAsAsset) {
+        return {
+            transaction_output_attribute: {
+                parent_transaction_id: nft.txid
+            },
+            transaction_data            : {
+                file_hash        : nft.hash,
+                attribute_type_id: 'Adl87cz8kC190Nqc'
+            },
+            transaction_data_type       : keepAsAsset ? 'tangled_asset' : 'transaction',
```

Comment:
let's create constants for every transaction_data_type and use them instead of using hardcoded strings all over the project. don't forget to also use these new consts when you call listTransactionWithDataReceived